radcad/core.py

```diff
@@ -164,6 +167,8 @@ def _single_run_wrapper(args):
         if raise_exceptions and exception:
             raise exception
         else:
+            if run_args.return_parameters:
+                [substep.update(run_args.parameters) for timestep in results for substep in timestep]
```

Comment:
One issue we might face including all parameters in the simulation results is that typically not all parameters are basic Python types, serialisable, or able to be stored in Pandas DataFrames efficiently.

As an alternative we do currently have access to each subset's parameters in the `before_subset()` and `after_subset()` hooks, should a user want to get access to these before or after a specific set of parameters is used in a simulation.

I understand the use case of storing parameters in the simulation results and having them end up in a DataFrame though - I have done this in the past for a subset of the parameters that are useful to have in post-processing and  maybe additionally are basic Python types.

Let me sleep on this and see if there are other ways of implementing, open to ideas too :)

Reply:
Fair point about complex data types for params. Given all params are specified as values in a dict though, I believe it should cover every case to output a dict with keys corresponding to run # and a value of a dict of params:
`{run_n: {params: { param_x: value_x }},
run_n+1: {params: { param_x: value_x }} }`

This approach would need to make `experiment.run()` return both the run results and the params dict at well. ie:
`results, params = experiment.run()`